src/ci/tested_versions_utils.py

```diff
@@ -1,38 +1,180 @@
 from __future__ import annotations
 
 import os
+import re
 from contextlib import contextmanager
 from dataclasses import dataclass
-from typing import Tuple
+from typing import List, Union
+
+# Major, minor, patch and (non semver standard, suffix)
+_semanticVersionPattern = re.compile(r'(!)?(\d+).(\d+).(\d+)([^\s]*)(?:\s*#\s*(.*))?')
+_splitVersionFromCommentPattern = re.compile(r'(!)?([^\s]*)(?:\s*#\s*(.*))?')
+
+# This file implements a custom version parsing and sorting mechanism,
+# as `packaging.version` has strange behaviors that won't work for other
+# languages, like:
+#
+# ```
+# >>> from packaging.version import parse
+# >>> str(parse("1.2.4c"))
+# '1.2.4rc0'
+# ```
+#
+# ```
+# >>> from packaging.version import parse
+# >>> str(parse("1.2.4b"))
+# '1.2.4b0'
+# ```
+
+
+@dataclass(frozen=True)
+class NonSemanticVersion:
+    supported: bool
+    version: str
+    comment: str
+
+    def __eq__(self, other):
+        if isinstance(other, SemanticVersion):
+           return False
+
+        return self.version == other.version
+
+    def __lt__(self, other):
+        if isinstance(other, SemanticVersion):
+           return True
+
+        return self.version < other.version
+
+
+@dataclass(frozen=True)
+class SemanticVersion:
+    supported: bool
+    major: int
+    minor: int
+    patch: int
+    suffix: str
+    comment: str
+
+    @property
+    def version(self):
+        return f"{self.major}.{self.minor}.{self.patch}{self.suffix or ''}"
+
+    def __eq__(self, other):
+        if isinstance(other, NonSemanticVersion):
+           return False
+
+        return (
+            self.major == other.major and
+            self.minor == other.minor and
+            self.patch == other.patch and
+            self.suffix == other.suffix
+        )
+
+    def __lt__(self, other):
+        if isinstance(other, NonSemanticVersion):
+           return False
+
+        if self.major < other.major:
+            return True
+
+        if self.major > other.major:
+            return False
+
+        if self.minor < other.minor:
+            return True
+
+        if self.minor > other.minor:
+            return False
+
+        if self.patch < other.patch:
+            return True
+
+        if self.patch > other.patch:
+            return False
+
+        if not self.suffix and self.suffix:
+            return True
+
+        if self.suffix and not self.suffix:
+            return False
+
+        return self.suffix < other.suffix
+
+
+def parse_version(version: str) -> Union[SemanticVersion, NonSemanticVersion]:
+    res = re.search(_semanticVersionPattern, version)
+
+    if res:
+        (supported, major, minor, patch, suffix, comment) = res.groups()
+        # The `supported` is either an empty string (supported) or the '!' string (not supported)
+        return SemanticVersion(not bool(supported), int(major), int(minor), int(patch), suffix, comment)
```

Comment:
```suggestion
        return SemanticVersion(not supported, int(major), int(minor), int(patch), suffix, comment)
```
The `bool` is redundant.